overlord/snapstate/target.go

```diff
@@ -110,7 +116,16 @@ func (t *target) snapsup(st *state.State, opts Options) (SnapSetup, error) {
 		PlugsOnly:          len(t.info.Slots) == 0,
 		InstanceKey:        t.info.InstanceKey,
 		ExpectedProvenance: t.info.SnapProvenance,
-	}, nil
+	}, compsups, nil
+}
+
+// componentTarget represents the data needed to setup a component for installation.
+type componentTarget struct {
+	// setup is a partially initialized ComponentSetup struct that contains the
+	// data needed to find the component that will be installed.
+	setup ComponentSetup
+	// info contains the snap.ComponentInfo for the component to be installed.
+	info *snap.ComponentInfo
```

Comment:
where is this field used?

Reply:
It can be eliminated in the current version of the code. I originally had it there because of how the `InstallGoal` implementation filled in the `ComponentSetup` struct. It partially filled it in, but that is no longer true. In that case, the entire `componentTarget` struct can go away.